Add DCAP evidence to BlockMetadataContents

nick-mobilecoin opened the pull request:
[rendered text](https://github.com/nick-mobilecoin/mcips/blob/nick/dcap-evidence-on-block-metadata/text/0068-dcap-evidence-in-block-metadata.md)

nick-mobilecoin committed:
Add DCAP evidence to BlockMetadataContents

eranrund commented on text/0068-dcap-evidence-in-block-metadata.md:
```diff
@@ -0,0 +1,92 @@
+attestation process. The `VerificationReport` contains the attestatoin evidence
+returned by the EPID attestation process.
+
+In general, block version 4 or greater is expected to only contain
```
The block version is a configuration parameter for consensus. As such, it is possible for us to launch a DCAP consensus with block version 3...
It seems like we'll need to update the consensus node configurations at the same time we are deploying the DCAP release, since it will start including the `dcap_evidence` in `BlockMetadataContents` and clients need to be prepared for that.
As such, we might want to consider pushing timestamps to block version 5 if we decide we don't want too many major changes in one release.

nick-mobilecoin replied:
At this time, due the new HW and significance of DCAP we're moving timestamps out to post 6.0 and thus they won't be on block 4.

eranrund commented on text/0068-dcap-evidence-in-block-metadata.md:
```diff
@@ -0,0 +1,92 @@
+
+While requiring a version bump, this change is able to be backwards compatible
+such that clients only knowing about version 3 can still consume the blocks.
+They will **not** be able to perform verification due to lacking the necessary
```
I think it's worth being more explicit here and state that they will not be able to verify the evidence, but they also will not be able to verify the `BlockMetadataContents` signature. I am not sure this makes the change backwards compatible...

nick-mobilecoin replied:
removed the "backwards compatible" wording and, moved the unable to verify wording into the `drawbacks` section.

nick-mobilecoin committed:
Remove wording of non breaking change

Moved mention of older clients being unable to verify the newer version
into the drawbacks section.
